apps/developer-portal/src/public/app-utils.js

```diff
@@ -0,0 +1,156 @@
+        _config = {};
+
+    return {
+        getConfig: function() {
+            return {
+                accountAppURL: _config.accountAppURL.origin + this.getTenantPath() + _config.accountAppURL.path,
+                appBase: _config.appBaseName,
+                appBaseWithTenant: this.getTenantPath() + "/" + _config.appBaseName,
+                clientID: (this.isSuperTenant()) ?
+                    _config.clientID : _config.clientID + "_" + this.getTenantName(),
+                clientOrigin: _config.clientOrigin,
+                clientOriginWithTenant: _config.clientOrigin + this.getTenantPath(),
+                debug: _config.debug,
+                loginCallbackURL: _config.clientOrigin + this.getTenantPath() + "/" + _config.appBaseName + 
+                    _config.loginCallbackPath,
+                logoutCallbackURL: _config.clientOrigin + this.getTenantPath() + "/" + _config.appBaseName + 
+                    _config.logoutCallbackPath,
+                routes: _config.routePaths,
+                serverOrigin: _config.serverOrigin,
+                serverOriginWithTenant: _config.serverOrigin + this.getTenantPath(),
+                tenant: (this.isSuperTenant()) ? this.getSuperTenant() : this.getTenantName(),
+                tenantPath: this.getTenantPath(),
+                ui: _config.ui
+            };
+        },
+
+        getSuperTenant: function() {
+            return _args.superTenant || "carbon.super";
+        },
+
+        getTenantName: function() {
+            var paths = window.location.pathname.split("/");
+            var tenantIndex = paths.indexOf(this.getTenantPrefix());
+
+            if (tenantIndex > 0) {
+                var tenantName = paths[tenantIndex + 1];
+                return (tenantName) ? tenantName : "";
+            } else {
+                return "";
+            }
+        },
+
+        getTenantPath: function() {
+            return (this.getTenantName() !== "") ?
+                "/" + this.getTenantPrefix() + "/" + this.getTenantName() : "";
+        },
+
+        getTenantPrefix: function() {
+            return _args.tenantPrefix || "t";
+        },
+
+        init: function(Args) {
+            _args = Args;
+
+            _default = {
+                "clientOrigin": window.location.origin,
+                "serverOrigin": _args.serverOrigin || "https://localhost:9443"
+            };
+
+            _config = _default;
+
+            var userConfigFile = (_config.clientOrigin + "/" + _args.deploymentConfigFile);
+
+            if (!userConfigFile) {
+                throw "AppUtils.init({deploymentConfigFile: value}) missing.";
+            } else {
+                loadUserConfig(userConfigFile, function(response) {
+                    var configResponse = JSON.parse(response);
+
+                    if (!{}.hasOwnProperty.call(configResponse, "accountAppURL"))
+                        throw "'accountAppURL' config is missing in " + _args.deploymentConfigFile;
```

Comment:
Maybe we can throw `IdentityAppsException` or extend it and create a new Exception for user config. WDYT?

Reply:
IdentityAppsException is having within the react app right? this execute before